src/script/standard.cpp

```diff
@@ -35,22 +35,49 @@ const char* GetTxnOutputType(txnouttype t)
     return nullptr;
 }
 
-bool Solver(const CScript& scriptPubKey, txnouttype& typeRet, std::vector<std::vector<unsigned char> >& vSolutionsRet)
+static bool MatchPayToPubkey(const CScript& script, valtype& pubkey)
 {
-    // Templates
-    static std::multimap<txnouttype, CScript> mTemplates;
-    if (mTemplates.empty())
-    {
-        // Standard tx, sender provides pubkey, receiver adds signature
-        mTemplates.insert(std::make_pair(TX_PUBKEY, CScript() << OP_PUBKEY << OP_CHECKSIG));
+    if (script.size() == 35 && script[0] == 33 && script[34] == OP_CHECKSIG) {
+        pubkey = valtype(script.begin() + 1, script.begin() + 34);
+        return CPubKey::ValidSize(pubkey);
+    }
+    if (script.size() == 67 && script[0] == 65 && script[66] == OP_CHECKSIG) {
+        pubkey = valtype(script.begin() + 1, script.begin() + 66);
+        return CPubKey::ValidSize(pubkey);
```

Comment:
Redundant check here? We know the size.

Reply:
We could just use proper constants in the lines above for more belt and suspenders.